tests/run-make/rust-lld/rmake.rs

```diff
@@ -0,0 +1,64 @@
+// Test linking using `cc` with `rust-lld`, using the unstable CLI described in MCP 510
+// see https://github.com/rust-lang/compiler-team/issues/510 for more info
+
+//@ needs-rust-lld
+//@ ignore-msvc
+//@ ignore-s390x lld does not yet support s390x as target
+
+extern crate run_make_support;
+
+use run_make_support::regex::Regex;
+use run_make_support::rustc;
+use std::process::Output;
+
+fn main() {
+    // Opt-in to lld and the self-contained linker, to link with rust-lld. We'll check that by
+    // asking the linker to display its version number with a link-arg.
+    let output = rustc()
+        .env("RUSTC_LOG", "rustc_codegen_ssa::back::link=info")
+        .arg("-Zlinker-features=+lld")
+        .arg("-Clink-self-contained=+linker")
+        .arg("-Zunstable-options")
+        .arg("-Clink-args=-Wl,-v")
```

Comment:
Could you introduce some weakly-typed helper methods for the stable flags? e.g.

https://github.com/rust-lang/rust/blob/023084804e5e8ea42877451c2b3030e7050281cc/src/tools/run-make-support/src/rustc.rs#L92-L96

so it could end up looking like

```rs
rustc()
    // ...
    .link_self_contained("+linker")
    .link_args("-Wl,-v")
```